wwwbase/js/scramble.js

```diff
@@ -0,0 +1,1051 @@
+      }
+      //$("#result").html(word.randomWord);
+      $("#maxWords").html(word.everyWord.length);
+      drawLetters(word.randomWord);
+      //Fire keyboard event checker
+      
+      //Start Timer
+      startTimer(difficulty);
+      //Fire Enter key checker
+      checkWord();
+
+      initLayerArrays();  
+      $(".wordArea").empty();
+      hide = 0; 
+
+      console.log(totalWords);
+      console.log(word.randomWord);
+
+    });
+  }
+
+  function initLayerArrays() {
+
+    layers = [];
+    upLayers = [];
+    downLayers = [];
+    lettersPressed = "";
+
+    layers = $("canvas").getLayers(); // Get drawn layers.
+    for(var i = 0 ; i < layers.length; i+= 2) //Select the layers with letter data.
+    {
+      upLayers.push(layers[i]);
+      downLayers.push(0);
+    }    
+  }
+
+
+
+  hide = 0;
+  function testing() {
+    $(".wordBtn").on("click", function() {
+      console.log("pushed");
+      $(this).blur();
+
+      var ul = 0;
+      var initialTR = "wordList";
+      var currentTR = initialTR;    
+      var start = 0;
+      var stop;
+
+      drawEnd();
+
+      if(!hide)
+      {
+        for(var i = 0; i <= totalWords.length; i++)
+        {
+          if( i % 5 == 0 || (i == totalWords.length && totalWords.length % 5 != 0))
+          {
+            stop = i;
+            var td = "td" + i;
+            var ulist = "ulist" + i;
+            //console.log("Appending td & ul");
+            $('<td></td>', { "class" : td }).appendTo("." + currentTR);
+            $('<ul></ul>', { "class" : ulist}).appendTo("." + td);
+            for(var k = start; k < stop; k++)
+            {
+              if(typeof totalWords[k] === "undefined")
+              {
+                break;
+              }
+              else
+              {
+                var list = "<li>" + totalWords[k] + "</li>";                          
+                $("." + ulist).append(list);                
+              }
+            }
+            ul++;
+            start = stop;
+          }
+          if($("." + currentTR).children().length % 9 == 0)
+          {
+            currentTR = initialTR + i;
+            $("<tr></tr>", {"class" : currentTR}).appendTo(".wordArea");
+
+          }        
+        }
+        hide = 1;
+      }
+      else
+      {
+        $(".wordArea").empty();
+        hide = 0;
+      }
+      /*
+      if ( !hide ) 
+      {
+        $(".wordArea").html(totalWords + " ");
+        hide = 1;
+  	  } 
+      else 
+      {
+          $(".wordArea").html("");
+          hide = 0;
+  	  }
+      */
+    });
+  }
+
+  function inputListen() {
+    //layers = $("canvas").getLayers();
+
+    $(document).keyup(function(letter) {
+      //var searchWord = $(this).val();
+      var key;
+      key = letter.keyCode;
+      var keyString;
+      keyString = String.fromCharCode(key);
+      //Transforma diacriticile in caractere fara diacritice.
+      for(var i = 0; i< layers.length; i += 2) {
```

Comment:
Eu schimb doar char code-ul preluat de la tastatura. In spate pe fiecare layer cuvintele sunt inca cu diacritice. Ca atare si cuvintele din baza sunt cautate si verificate tot cu diacritice( daca exista).